dbms/src/Interpreters/WindowDescription.cpp

```diff
@@ -52,18 +54,31 @@ WindowFrame::FrameType getFrameTypeFromTipb(const tipb::WindowFrameType & type)
     }
 }
 
-void WindowDescription::setWindowFrame(const tipb::WindowFrame & frame_)
+void setWindowFrameImpl(WindowFrame & frame, const tipb::WindowFrame & tipb_frame)
 {
-    frame.type = getFrameTypeFromTipb(frame_.type());
-    frame.begin_offset = frame_.start().offset();
-    frame.begin_type = getBoundaryTypeFromTipb(frame_.start());
-    frame.begin_preceding = (frame_.start().type() == tipb::WindowBoundType::Preceding);
-    frame.begin_cmp_data_type = frame_.start().cmp_data_type();
-    frame.end_offset = frame_.end().offset();
-    frame.end_type = getBoundaryTypeFromTipb(frame_.end());
-    frame.end_preceding = (frame_.end().type() == tipb::WindowBoundType::Preceding);
-    frame.end_cmp_data_type = frame_.end().cmp_data_type();
+    frame.type = getFrameTypeFromTipb(tipb_frame.type());
+    frame.begin_offset = tipb_frame.start().offset();
+    frame.begin_type = getBoundaryTypeFromTipb(tipb_frame.start());
+    frame.begin_preceding = (tipb_frame.start().type() == tipb::WindowBoundType::Preceding);
+    frame.begin_cmp_data_type = tipb_frame.start().cmp_data_type();
+    frame.end_offset = tipb_frame.end().offset();
+    frame.end_type = getBoundaryTypeFromTipb(tipb_frame.end());
+    frame.end_preceding = (tipb_frame.end().type() == tipb::WindowBoundType::Preceding);
+    frame.end_cmp_data_type = tipb_frame.end().cmp_data_type();
     frame.is_default = false;
+
+    if (frame.type == WindowFrame::FrameType::Rows)
+    {
+        if (frame.begin_type == WindowFrame::BoundaryType::Offset && frame.begin_offset == 0)
+            frame.begin_type = WindowFrame::BoundaryType::Current;
```

Comment:
I think you also need to adjust `begin_preceding` and `end_preceding`, please double comfirm it.

Reply:
> I think you also need to adjust `begin_preceding` and `end_preceding`, please double comfirm it.

I think it's needless to adjust `begin_preceding`

Reply:
in my understanding, if the tipb_frame is 0 following, then the preceding will be false, but if the tipb_frame is current_row, preceding will be true, you adjust the type, but does not adjust the preceding, which will makes it in-consistent

Reply:
But `WindowFrame`'s `==` operator will always check preceding
https://github.com/pingcap/tiflash/blob/bf34f36639972a663c9616e8a999fca60be8b525/dbms/src/Interpreters/WindowDescription.h#L96-L103

Reply:
> But `WindowFrame`'s `==` operator will always check preceding
> 
> https://github.com/pingcap/tiflash/blob/bf34f36639972a663c9616e8a999fca60be8b525/dbms/src/Interpreters/WindowDescription.h#L96-L103

adjusted